repository/src/main/java/org/alfresco/service/cmr/search/CategoryService.java

```diff
@@ -234,15 +236,27 @@ public Collection<ChildAssociationRef> getRootCategories(StoreRef storeRef, QNam
      */
     @Auditable(parameters = {"nodeRef"})
     public void deleteCategory(NodeRef nodeRef);
-    
-   /** 
-    * Get the most polular categories 
-    * 
+
+   /**
+    * Get the most polular categories
+    *
     * @param storeRef StoreRef
     * @param aspectName QName
     * @param count int
     * @return List
     */
     @Auditable(parameters = {"storeRef", "aspectName", "count"})
     public List<Pair<NodeRef, Integer>> getTopCategories(StoreRef storeRef, QName aspectName, int count);
+
+    /**
+     * Get a root category NodeRef
+     *
+     * @return NodeRef for category root node
+     */
+    @Experimental
+    @Auditable()
+    default Optional<NodeRef> getRootCategoryNodeRef(final StoreRef storeRef)
+    {
+        return Optional.empty();

```

Comment:
Why `empty` is returned here?

Reply:
I think this is always better that returning null or not implemented. 
This is also needed because the interface is implemented in ent-repo as well.

Reply:
This method is not yet used (I extracted it from other layer - `Categories` interface - when we discussed how root category should be handled) since my latest changes.
I am beginning to wonder if this method will still be needed but will keep it at least until we're much farther with the whole implementation.

Reply:
I'm ok with this explanation.